core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.component.ts

```diff
@@ -173,7 +224,119 @@ export class TranslationModalComponent {
     this.activeWrittenTranslation.html = textAndAvailability.translationHtml;
   }
 
+  getElementAttributeTexts(elements: HTMLElement[], type: string): string[] {
+    const textWrapperLength = 6;
+    const attributes = Array.from(elements, function(element: HTMLElement) {
+      // A sample element would be as <oppia-noninteractive-image alt-with-value
+      // ="&amp;quot;Image description&amp;quot;" caption-with-value=
+      // "&amp;quot;Image caption&amp;quot;" filepath-with-value="&amp;quot;
+      // img_20210129_210552_zbv0mdty94_height_54_width_490.png&amp;quot;">
+      // </oppia-noninteractive-image>
+      if (element.localName === 'oppia-noninteractive-image') {
+        const attribute = element.attributes[type].value;
+        return attribute.substring(
+          textWrapperLength, attribute.length - textWrapperLength);
+      }
+    });
+    return attributes.filter(attribute => attribute);
+  }
+
+  getImageAttributeTexts(htmlElements: HTMLElement[]): ImageDetails {
+    return {
+      filePaths: this.getElementAttributeTexts(
+        htmlElements, 'filepath-with-value'),
+      alts: this.getElementAttributeTexts(
+        htmlElements, 'alt-with-value'),
+      descriptions: this.getElementAttributeTexts(
+        htmlElements, 'caption-with-value')
+    };
+  }
+
+  copiedAllElements(
+      originalElements: string[],
+      translatedElements: string[]): boolean {
+    const hasMatchingTranslatedElement = (element) => (
+      translatedElements.includes(element));
+    return originalElements.every(hasMatchingTranslatedElement);
+  }
+
+  hasSomeDuplicateElements(
+      originalElements: string[],
+      translatedElements: string[]): boolean {
+    if (originalElements.length === 0) {
+      return false;
+    }
+    const hasMatchingTranslatedElement = (element) => (
+      translatedElements.includes(element) && originalElements.length > 0);
+    return originalElements.some(hasMatchingTranslatedElement);
+  }
+
+  isTranslationCompleted(
+      originalElements: HTMLElement[],
+      translatedElements: HTMLElement[]): boolean {
+    const filteredOriginalElements = originalElements.filter(
+      (element: HTMLElement) => element.nodeType === Node.ELEMENT_NODE);
```

Comment:
Type declarations seem unnecessary for arrow functions. Ditto below. 

```suggestion
      element => element.nodeType === Node.ELEMENT_NODE);
```

Reply:
Done